src/Microsoft.AspNet.Razor.Runtime/TagHelpers/TagHelperDescriptorResolver.cs

```diff
@@ -124,6 +133,79 @@ public IEnumerable<TagHelperDescriptor> Resolve([NotNull] TagHelperDescriptorRes
             return descriptors;
         }
 
+        private static IEnumerable<TagHelperDescriptor> PrefixDescriptors(
+            TagHelperDescriptorResolutionContext context,
+            IEnumerable<TagHelperDescriptor> descriptors)
+        {
+            var tagHelperPrefix = ResolveTagHelperPrefix(context);
+
+            if (!string.IsNullOrEmpty(tagHelperPrefix))
+            {
+                return descriptors.Select(descriptor =>
+                    new TagHelperDescriptor(
+                        tagHelperPrefix,
+                        descriptor.TagName,
+                        descriptor.TypeName,
+                        descriptor.AssemblyName,
+                        descriptor.Attributes));
+            }
+
+            return descriptors;
+        }
+
+        private static string ResolveTagHelperPrefix(TagHelperDescriptorResolutionContext context)
+        {
+            var prefixDirectiveDescriptors = context.DirectiveDescriptors.Where(
+                descriptor => descriptor.DirectiveType == TagHelperDirectiveType.TagHelperPrefix);
+
+            // For each invalid @tagHelperPrefix we need to create an error.
+            foreach (var erroredContext in prefixDirectiveDescriptors.Skip(1))
```

Comment:
`erroredContext` -> `directive` ("errored" isn't a word and the variable doesn't hold a context)